src/diffcalc/common_types.py

```diff
@@ -0,0 +1,17 @@
+"""Common types used throughout diffcalc-core, required by diffcalc_API."""
+
+from dataclasses import dataclass
+
+
+@dataclass
```

Comment:
I see we have a mix of dataclasses and pydantic models. Was there a specific motivation for this? I think it's best to have all of one thing if possible.